research/src/pages/file.proposal.mdx

```diff
@@ -0,0 +1,165 @@
+---
+menu: Components
+name: File (Editor's Draft)
+path: /components/file
+pathToResearch: /components/file.research
+---
+
+# File Input Component Specification
+
+## Overview <a href="#overview" id="overview"></a>
+
+The File Input component is a control that provides the user with the ability
+to select local files.
+
+### Background <a href="#background" id="background"></a>
+
+File Input controls are widely used and implemented on the web. They are as of
+today, the only thing that allows an application to access the local file
+system, in a fairly limited but controlled way.
+
+### Use Cases <a href="#use-cases" id="use-cases"></a>
+
+The `<input type=file>` control is widely used to provide applications with the
+capability of reading local files on form submission, or uploading files to a
+server.
+
+### Features <a href="#features" id="features"></a>
+
+The main feature is selecting local files, potentially receiving hints from the
+website about which file kinds are allowed, for example.
+
+### Prior Art/Examples <a href="#prior-art" id="prior-art"></a>
+
+#### Browser implementations
+
+Browsers show an interoperable rendering for file inputs, consisting of a
+button and a label.
+
+- WebKit and Blink use an `<input type=button>`, and a magical label that
+  doesn't show up in the DOM.
+
+- Gecko uses an actual `<button tabindex=-1>` and an HTML `<label>` element.
+
+Those three implementations use some form of middle-cropping when the selected
+file name is too large.
+
+#### Custom inputs on the web
+
+Custom inputs on the web are generally achieved by hiding the original input.
+Some approach it by hiding it in a way that the user can still hit (like
+`opacity: 0`), others use JS to forward the click from some other UI element to
+the underlying file input.
+
+- [Bootstrap](https://getbootstrap.com/docs/4.0/components/forms/#file-browser)
+  uses the former approach, by generating a "Browse" button with generated
+  content.
+
+- [Carbon Design System](https://www.carbondesignsystem.com/components/file-uploader/code/)
+  uses the later.
+
+---
+
+### API <a href="#api" id="api"></a>
+
+See [the Resources section](#resources) for the specification and the MDN reference.
+
+#### Properties and Attributes <a href="#properties-attributes" id="properties-attributes"></a>
+
+See [the Resources section](#resources) for the specification and the MDN reference.
```

Comment:
I'd prefer for this to be filled in. That said - I'd also love to chat with the MDN team to see if they have an API endpoint we can possibly pull this in from on build.

Reply:
@chrisdavidmills do you know if MDN has some API we could use here?

(I'd rather punt on this until we have the setup figured out, but let me know if you want to go through the whole thing... Not all `<input>` APIs apply to `type=file` I guess.)

Reply:
I'm ok punting on it for now but can you file an issue for it then?

Reply:
We have a separate reference doc for each input type — see [`<input type="file">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file), and a [section here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement#Properties_file) that lists properties of the `HTMLInputElement` interface that apply only to file types.

But we don't have an API that would expose such info automatically, as such. I'd be interested to hear more info about what you'd want here.

Reply:
@chrisdavidmills to avoid discussing this in the PR is there a place we can have this discussion. Possibly on an MDN github issue?

Reply:
@gregwhitworth yup — best place is an issue on https://github.com/mdn/sprints/issues/